main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -417,6 +424,19 @@ object Kinder {
       val exp2 = visitExp(exp20, kenv0, taenv, henv0, root)
       KindedAst.Expr.LetRec(sym, ann, mod, exp1, exp2, loc)
 
+    case ResolvedAst.Expr.LocalDef(ann, sym, fparams0, dtpe0, deff0, exp10, exp20, loc) =>
```

Comment:
I've never looked at the kinder before, maybe @mlutze can have a guest appearance 😄

Reply:
I think this looks reasonable (I haven't looked at Kinder in a long time)

But we should add a test or two since this is a new special case.